File: PageObjectJava/src/test/java/PageObjects/AutomationPracticeStore/Objects/NewAccountEmail.java
package PageObjects.AutomationPracticeStore.Objects;

import Base.TestConfiguration;
import PageObjects.AutomationPracticeStore.ErrorMessageException;
import PageObjects.AutomationPracticeStore.NewAccountForm;
import org.openqa.selenium.By;
import org.openqa.selenium.TimeoutException;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.ArrayList;
import java.util.List;

public class NewAccountEmail {

    @FindBy(how = How.ID_OR_NAME, using = "email_create")
    private WebElement inputCreateAccountEmailAddress;

    @FindBy(how = How.ID_OR_NAME, using = "SubmitCreate")
    private WebElement buttonCreateAccount;


    public NewAccountEmail setEmailAddress(String emailAddress){
        inputCreateAccountEmailAddress.clear();
        inputCreateAccountEmailAddress.sendKeys(emailAddress);
        return this;
    }

    public String getEmailAddress(){
        return inputCreateAccountEmailAddress.getAttribute("value");
    }

    public NewAccountForm pressCreateAccountButton(){
        WebDriverWait wait = new WebDriverWait(TestConfiguration.getDriver(), 2);
        buttonCreateAccount.click();
        try{
            wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("create_account_error")));
        }catch (TimeoutException ex){
            return new NewAccountForm();
        }
        List<String> errorsFound = new ArrayList<>();
        for(WebElement errorWe : TestConfiguration.getDriver().findElements(By.cssSelector("div#create_account_error li")))
            errorsFound.add(errorWe.getText());
        throw new ErrorMessageException(errorsFound);
    }

}

File: PageObjectPython/PageObjects/AutomationPracticeStore/Objects/NewAccountEmail.py
from selenium.common.exceptions import TimeoutException
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support.expected_conditions import visibility_of_element_located

from PageObjects.AutomationPracticeStore.NewAccountForm import NewAccountForm
from PageObjects.AutomationPracticeStore.Objects.POExceptions import ErrorMessageException
from PageObjects.BasePageObject import BasePageObject


class NewAccountEmail(BasePageObject):

    __input_create_account_email_address = (By.ID, 'email_create')
    __button_create_account = (By.ID, 'SubmitCreate')

    def set_email_address(self, email_address):
        input_box = self.base_page_driver.find_element(*self.__input_create_account_email_address)
        input_box.clear()
        input_box.send_keys(email_address)
        return self

    def get_email_address(self):
        return self.base_page_driver.find_element(*self.__input_create_account_email_address).get_attribute('value')

    def press_create_account_button(self):
        wait = WebDriverWait(self.base_page_driver, 2)
        self.base_page_driver.find_element(*self.__button_create_account).click()
        try:
            wait.until(visibility_of_element_located((By.ID, 'create_account_error')))
        except TimeoutException:
            return NewAccountForm()
        else:
            errors_found = []
            for error_element in self.base_page_driver.find_elements((By.CSS_SELECTOR, 'div#create_account_error li')):
                errors_found.append(str(error_element.text))
            raise ErrorMessageException.by_errors_list(errors_found)
